examples/tv-app/linux/include/application-basic/ApplicationBasicManager.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ApplicationBasicManager.h"
+
+#include "gen/attribute-id.h"
+#include "gen/attribute-type.h"
+#include "gen/cluster-id.h"
+#include "gen/command-id.h"
+
+#include <app/util/af.h>
+#include <app/util/basic-types.h>
+
+#include <inipp/inipp.h>
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+using namespace std;
+
+CHIP_ERROR ApplicationBasicManager::Init()
+{
+    CHIP_ERROR err                                       = CHIP_NO_ERROR;
+    EndpointConfigurationStorage & endpointConfiguration = EndpointConfigurationStorage::GetInstance();
+    err                                                  = endpointConfiguration.Init();
+    SuccessOrExit(err);
+    es = &endpointConfiguration;
+exit:
+    return err;
+}
+
+void ApplicationBasicManager::store(chip::EndpointId endpoint, Application * application)
+{
+    if (strcmp(application->vendorName, ""))
+    {
+        EmberAfStatus vendorNameStatus =
+            emberAfWriteServerAttribute(endpoint, ZCL_APPLICATION_BASIC_CLUSTER_ID, ZCL_APPLICATION_VENDOR_NAME_ATTRIBUTE_ID,
+                                        (uint8_t *) &application->vendorName, ZCL_CHAR_STRING_ATTRIBUTE_TYPE);
+        if (vendorNameStatus != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            std::cerr << "Failed to store attribute. ERR:";
```

Comment:
Is there supposed to be something printed after `ERR:` here ?

Can we use the logging APIs instead?

Reply:
Updated to emberAfPrintf.